src/infra/src/table/search_queue.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use sea_orm::{
+    entity::prelude::*, ColumnTrait, ConnectionTrait, DatabaseBackend, EntityTrait, PaginatorTrait,
+    QueryFilter, Schema, Set,
+};
+
+use super::get_lock;
+use crate::{
+    db::{connect_to_orm, mysql, postgres, sqlite, IndexStatement, ORM_CLIENT},
+    errors,
+};
+
+// define the short_urls table
+#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
+#[sea_orm(table_name = "search_queue")]
+pub struct Model {
+    #[sea_orm(primary_key, auto_increment = true)]
+    pub id: i64,
+    #[sea_orm(column_type = "String(StringLen::N(16))")]
+    pub work_group: String,
+    #[sea_orm(column_type = "String(StringLen::N(256))")]
+    pub user_id: String,
+    #[sea_orm(column_type = "String(StringLen::N(32))")]
+    pub trace_id: String,
+    pub created_ts: i64,
+}
+
+#[derive(Copy, Clone, Debug, EnumIter)]
+pub enum Relation {}
+
+impl RelationTrait for Relation {
+    fn def(&self) -> RelationDef {
+        panic!("No relations defined")
+    }
+}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+pub async fn init() -> Result<(), errors::Error> {
+    create_table().await?;
+    create_table_index().await?;
+    Ok(())
+}
```

Comment:
I'm concerned that if `Model` ever changes then this will change the behavior of the migration script. This could put us in the risky position where a user that run the migration script before the change and a user that runs the migration script after the change would have different database schemas despite both seemingly having ran all pending migrations. 

We have actually already ran into this problem with this [PR](https://github.com/openobserve/openobserve/pull/5232). A change was made to the `Model` to fix an issue with the migration, but since the migration had already ran successfully on our alpha1 environment - our alpha1 environment now has a DB schema that differs from environments with "up to date migrations". More details here:  https://github.com/openobserve/openobserve/issues/5233